packages/grid/_modules_/grid/components/cell/GridCell.tsx

```diff
@@ -145,13 +145,17 @@ export const GridCell = React.memo(function GridCell(props: GridCellProps) {
   };
 
   React.useLayoutEffect(() => {
-    const doc = ownerDocument(apiRef!.current.rootElementRef!.current as HTMLElement);
+    if (!hasFocus) {
+      return;
+    }
+
+    if (cellMode === 'edit' && isEditable) {
```

Comment:
Updated. Answering your question, if `apiRef.current.setCellMode` is called in a cell that cannot be edited it will turn to edit mode. We need to add a check there.